framework/src/problems/EigenProblem.C

```diff
@@ -346,6 +347,29 @@ EigenProblem::solve()
     TIME_SECTION(_solve_timer);
     _nl->solve();
     _nl->update();
+
+    if (_has_normalization)
```

Comment:
Could you move  the `if` block  as an independent function? And we call that function from here.

I personally like modular